compiler+runtime/src/cpp/jank/runtime/obj/tagged_literal.cpp

```diff
@@ -0,0 +1,99 @@
+#include <jank/runtime/obj/symbol.hpp>
+#include <jank/runtime/core/to_string.hpp>
+#include <jank/runtime/visit.hpp>
+#include <jank/runtime/context.hpp>
+#include <jank/runtime/obj/tagged_literal.hpp>
+
+namespace jank::runtime::obj
+{
+  tagged_literal::tagged_literal(object_ptr const tag, object_ptr const form)
+    : tag{ tag }
+    , form{ form }
+  {
+  }
+
+  native_bool tagged_literal::equal(object const &o) const
+  {
+    if(o.type != object_type::symbol)
+    {
+      return false;
+    }
+
+    auto const s(expect_object<tagged_literal>(&o));
+    return tag == s->tag && form == s->form;
+  }
+
+  static void
+  to_string_impl(object_ptr const &tag, object_ptr const &form, util::string_builder &buff)
+  {
+    buff('#');
+    to_string(tag, buff);
+    buff(' ');
+    to_string(form, buff);
+  }
+
+  void tagged_literal::to_string(util::string_builder &buff) const
+  {
+    to_string_impl(tag, form, buff);
+  }
+
+  native_persistent_string tagged_literal::to_string() const
+  {
+    util::string_builder buff;
+    to_string_impl(tag, form, buff);
+    return buff.release();
+  }
+
+  native_persistent_string tagged_literal::to_code_string() const
+  {
+    return to_string();
+  }
+
+  native_hash tagged_literal::to_hash() const
+  {
+    if(hash)
+    {
+      return hash;
+    }
+
+    return hash = hash::combine(hash::visit(tag), hash::visit(form));
+  }
+
+  object_ptr tagged_literal::get(object_ptr const key, object_ptr const fallback) const
+  {
+    object_ptr _tag = __rt_ctx->intern_keyword("tag").expect_ok();
+    object_ptr _form = __rt_ctx->intern_keyword("form").expect_ok();
+
+    if(_tag == key)
+    {
+      return tag;
+    }
+
+    if(_form == key)
+    {
+      return form;
+    }
+
+    return fallback;
+  }
+
+  object_ptr tagged_literal::get(object_ptr const key) const
+  {
+    return get(key, obj::nil::nil_const());
+  }
+
+  object_ptr tagged_literal::get_entry(object_ptr /*key*/) const
```

Comment:
I don't think `get_entry` and `contains` are supported by `tagged-literals` on the jvm, so not sure what to do here.

Reply:
`get_entry` returns a vector of key and value. Something like:

```cpp
return make_box<persistent_vector>(std::in_place, tag, form);
```